libvips/foreign/magick2vips.c

```diff
@@ -385,6 +386,25 @@ parse_header(Read *read)
 		return -1;
 	}
 
+	switch (image->units) {
+	case PixelsPerInchResolution:
+		im->Xres = image->x_resolution / 25.4;
+		im->Yres = image->y_resolution / 25.4;
+		break;
+
+	case PixelsPerCentimeterResolution:
+		im->Xres = image->x_resolution / 10.0;
+		im->Yres = image->y_resolution / 10.0;
+		break;
+
+	default:
+		im->Xres = 1.0;
+		im->Yres = 1.0;
+		break;
+	}
+
+	// this can be wrong for some GM versions and must be sanity checked (see
+	// below)
```

Comment:
I still have to get used to the fact that we now support C++ style `//` comments. :)